tidy3d/components/data/monitor_data.py

```diff
@@ -992,6 +1040,12 @@ def overlap_sort(
             modes is less than this threshold, a warning about a possible discontinuity is
             displayed.
         """
+        if len(self.field_components) == 0:
+            # raise DataError(
```

Comment:
resolve comment? (remove or discuss?)

Reply:
Yes, I feel the error would be more in line to other functions (like trying to calculate intensity without the fields present), but because we only return a copy of the original data with modes reordered, it makes sense to order an empty set of modes and not bother the user with checking if the fields exist before calling the method. So I prefer to return a copy, but I left the option to raise an error if that's preferable.